tools/extra/fpgabist/dma/fpga_dma.c

```diff
@@ -466,48 +783,81 @@ static fpga_result _write_memory_mmio_unaligned(fpga_dma_handle dma_h, uint64_t
 *
 * @brief                   Writes to a DWORD/QWORD aligned memory address(FPGA address).
 * @param[in] dma           Handle to the FPGA DMA object
-* @param[in/out] dst_ptr   FPGA address
-* @param[in/out] src_ptr   Host buffer address
-* @param[in/out] count     Size in bytes
-* @return fpga_result      FPGA_OK on success, return code otherwise
+* @param[in/out] dst_ptr   Pointer to the FPGA address
+* @param[in/out] src_ptr   Pointer to the Host buffer address
+* @param[in/out] count     Pointer to the Size in bytes
+* @return fpga_result      FPGA_OK on success, return code otherwise.  Updates src, dst, and count
 *
 */
-static fpga_result _write_memory_mmio(fpga_dma_handle dma_h, uint64_t *dst_ptr, uint64_t *src_ptr, uint64_t *count)
+static fpga_result _write_memory_mmio(fpga_dma_handle dma_h, uint64_t * dst_ptr,
+				      uint64_t * src_ptr, uint64_t * count)
 {
 	fpga_result res = FPGA_OK;
+
+	if (*count < DWORD_BYTES)
+		return res;
+
+	assert(*count >= DWORD_BYTES);
+	assert(IS_ALIGNED_DWORD(*dst_ptr));
+	if (!IS_ALIGNED_DWORD(*dst_ptr))	// If QWORD aligned, this will be true
+		return FPGA_EXCEPTION;
```

Comment:
Line 803 is unreachable code. If *dst_ptr is not DWORD_ALIGNED, the assertion on line 801 will fail causing the program to abort.